### PR TITLE
Add support for Media Queries to stylex shim

### PR DESCRIPTION
**Pull request by necolas**
Updates the Media Query support to modern stylex syntax.

Plus 25-35% faster results for `css.props` benchmarks.

### CODE REVIEW

**Comment by necolas** [comment]
## RSD benchmarks (for native)

Summary of https://github.com/facebook/react-strict-dom/pull/149#issuecomment-2184246539

Suite | BASE | PATCH | Ratio
-- | -- | -- | --
css.create |   |   |  
small | 1,167,997 | 1,188,921 | 1.02
small with units | 442,877 | 445,691 | 1.01
small with variables | 677,006 | 666,689 | 0.98
several small | 320,178 | 331,083 | 1.03
large | 214,661 | 219,547 | 1.02
large with polyfills | 145,167 | 144,538 | 1.00
complex | 107,111 | 111,342 | 1.04
unsupported | 207,221 | 209,710 | 1.01
css.createTheme |   |   |  
simple theme | 221,895 | 221,963 | 1.00
polyfill theme | 209,419 | 210,643 | 1.01
css.props |   |   |  
small | 172,076 | 226,305 | 1.32
small with units | 147,446 | 195,312 | 1.32
small with variables | 78,559 | 105,438 | 1.34
small with variables of units | 59,979 | 77,038 | 1.28
large | 68,406 | 89,972 | 1.32
large with polyfills | 20,398 | 25,314 | 1.24
complex | 15,876 | 19,104 | 1.20
unsupported | 53,793 | 66,298 | 1.23
simple merge | 139,612 | 182,330 | 1.31
wide merge | 11,678 | 14,569 | 1.25
deep merge | 11,586 | 14,387 | 1.24
internals |   |   |  
extractStyleThemes | 461,103 | 462,212 | 1.00
flattenStyle | 652,544 | 653,992 | 1.00

File: packages/react-strict-dom/tests/css-test.native.js [hunk 1]
```diff
@@ -1610,9 +1611,9 @@ expect.extend({
     const EXPECTED_MATCHED_VALUE = 500;
     const { underTest } = css.create({
       underTest: {
-        width: UNEXPECTED_MATCHED_VALUE,
-        [`@media ${query}`]: {
-          width: EXPECTED_MATCHED_VALUE
+        width: {
+          default: UNEXPECTED_MATCHED_VALUE,
+          [`${query}`]: EXPECTED_MATCHED_VALUE
```
Comment (nmn) [inline packages/react-strict-dom/tests/css-test.native.js:1616]:
```suggestion
          [query]: EXPECTED_MATCHED_VALUE
```

**Review by nmn — APPROVED** [review]

**Comment by nmn** [comment]
Does this also handle `:hover`, `:focus` and `:active` in the new syntax?

**Comment by necolas** [comment]
Merged c828b742ddbec250daafe6c9e94c0313e1adf5e4

**Pull request closed without merging**